lib/Detector.js

```diff
@@ -104,28 +135,11 @@ export default class Detector {
         return;
       }
 
-      for (const [feat] of Object.entries(this.features).filter(([, featValue]) => {
-        if (!featValue) return false;
-        if (typeof featValue === 'function') {
-          return featValue(child);
-        }
-        if (Array.isArray(featValue)) {
-          return featValue.some((function_) => function_(child));
-        }
-        if (child.type !== 'decl') {
-          return false;
-        }
-
-        return Object.entries(featValue).some(([property, value]) => {
-          if (property !== '' && property !== child.prop) return false;
-          if (value === true) return true;
-          if (value === false) return false;
-          return performFeatureCheck(value, stripUrls(child.value));
-        });
-      })) {
-        const feature = /** @type {keyof FEATURES} */ (feat);
-        callback({ usage: child, feature, ignore: this.ignore });
+      const detectedFeatures = this.features.filter(([, ruleCheck]) => ruleCheck(child));
```

Comment:
Couldn't we save some time here by not double iterating (once on `this.features` and again on the filtered list)?